irctest/server_tests/test_message_tags.py

```diff
@@ -154,25 +154,30 @@ def testLengthLimits(self):
         self.assertEqual(len(max_privmsg), 4096 + (512 - 2))
         self.sendLine("alice", max_privmsg)
         echo = self.getMessage("alice")
-        relay = self.getMessage("bob")
-        self.assertMessageMatch(
-            echo,
-            command="PRIVMSG",
-            params=["#test", StrRe("b{400,496}")],
-            tags={"+baz": "a" * 4081, "msgid": StrRe(".+"), **ANYDICT},
-        )
-        self.assertMessageMatch(
-            relay,
-            command="PRIVMSG",
-            params=["#test", StrRe("b{400,496}")],
-            tags={"+baz": "a" * 4081, "msgid": StrRe(".+"), **ANYDICT},
-        )
-        self.assertEqual(echo.tags["msgid"], relay.tags["msgid"])
-        # message may have been truncated
-        self.assertIn("b" * 400, echo.params[1])
-        self.assertEqual(echo.params[1].rstrip("b"), "")
-        self.assertIn("b" * 400, relay.params[1])
-        self.assertEqual(relay.params[1].rstrip("b"), "")
+        # the server may still reject this message on the grounds that the final
+        # parameter is too long to be relayed without truncation, once alice's
+        # NUH is included. however, if the message was accepted, the tags MUST
+        # be relayed intact, because they are unquestionably valid:
```

Comment:
Add a link to https://defs.ircdocs.horse/defs/numerics.html#err-inputtoolong-417 and add `@cases.mark.specifications("ircdocs")`